custom/crediblex/portfolio/loc/src/main/java/com/crediblex/fineract/portfolio/loc/service/LineOfCreditWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,255 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.crediblex.fineract.portfolio.loc.service;
+
+import com.crediblex.fineract.portfolio.loc.domain.LineOfCredit;
+import com.crediblex.fineract.portfolio.loc.repository.LineOfCreditRepository;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.fineract.infrastructure.core.api.JsonCommand;
+import org.apache.fineract.infrastructure.core.data.CommandProcessingResult;
+import org.apache.fineract.infrastructure.core.data.CommandProcessingResultBuilder;
+import org.apache.fineract.infrastructure.core.exception.PlatformApiDataValidationException;
+import org.apache.fineract.infrastructure.security.service.PlatformSecurityContext;
+import org.apache.fineract.portfolio.client.domain.Client;
+import org.apache.fineract.portfolio.client.domain.ClientRepositoryWrapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Slf4j
+public class LineOfCreditWritePlatformServiceImpl implements LineOfCreditWritePlatformService {
+
+    private final PlatformSecurityContext context;
+    private final LineOfCreditRepository lineOfCreditRepository;
+    private final ClientRepositoryWrapper clientRepository;
+    private final LineOfCreditDataValidator dataValidator;
+
+    @Autowired
+    public LineOfCreditWritePlatformServiceImpl(PlatformSecurityContext context, LineOfCreditRepository lineOfCreditRepository,
+            ClientRepositoryWrapper clientRepository, LineOfCreditDataValidator dataValidator) {
+        this.context = context;
+        this.lineOfCreditRepository = lineOfCreditRepository;
+        this.clientRepository = clientRepository;
+        this.dataValidator = dataValidator;
+    }
+
+    @Override
+    @Transactional
+    public CommandProcessingResult createLineOfCredit(JsonCommand command) {
```

Comment:
Since you have a typed object at the API layer, i think you should also pass a typed object here to avoid manually reading values from the map. You can do this by genering the object from the NewCommandHandler interface with gson or jackson